src/Distribution/Nixpkgs/Meta.hs

```diff
@@ -67,34 +105,49 @@ instance Pretty Meta where
     [ onlyIf (not (null _homepage)) $ attr "homepage" $ string _homepage
     , onlyIf (not (null _description)) $ attr "description" $ string _description
     , attr "license" $ pPrint _license
-    , onlyIf (_platforms /= allKnownPlatforms) $ renderPlatforms "platforms" _platforms
-    , onlyIf (_hydraPlatforms /= _platforms) $ renderPlatforms "hydraPlatforms" _hydraPlatforms
+    , maybe mempty (renderPlatforms "platforms") _platforms
+    , maybe mempty (renderPlatforms "badPlatforms") _badPlatforms
+    , maybe mempty (renderPlatforms "hydraPlatforms") _hydraPlatforms
     , setattr "maintainers" (text "with lib.maintainers;") (Set.map (view ident) _maintainers)
     , boolattr "broken" _broken _broken
     ]
 
-renderPlatforms :: String -> Set Platform -> Doc
+partitionEithers :: (Ord a, Ord b) => Set (Either a b) -> (Set a, Set b)
+partitionEithers s =
+  let (a, b) = Set.partition isLeft s
+  in (Set.map (fromLeft undefined) a, Set.map (fromRight undefined) b)
+
+renderPlatforms :: String -> Set NixpkgsPlatform -> Doc
```

Comment:
Could you add a few examples to the documentation of this function of how different values for `Set NixpkgsPlatform` end up getting rendered?

Reply:
Done in f92d219d6661f9f9234b06d7ce51d41c88f6fd7f. Will flesh out the `Meta` example as well as soon as we have finalized this (remind me).

Reply:
Thanks, this looks nice